lib/rspec/mocks/object_reference.rb

```diff
@@ -70,11 +74,15 @@ def when_loaded
     # Represents a reference to that module.
     #
     # @private
-    class DirectModuleReference < DirectObjectReference
+    class AnonymousModuleReference < DirectObjectReference
```

Comment:
The documentation comment for this class is still inaccurate ("Used when a module is passed to `class_double` or `instance_double`" -- it's actually only used when an anonymous module is passed) -- but that's not a merge blocker.  I'll plan to get to it post-merge if you don't get to it first.